openmeter/app/stripe/entity/app/app_invoice.go

```diff
@@ -0,0 +1,426 @@
+			for _, discount := range discounts {
+				// Add line item if it doesn't exist
+				if line.ExternalIDs.Invoicing == "" {
+					lineIDs = append(lineIDs, line.ID)
+
+					addStripeLines.Lines = append(addStripeLines.Lines, &stripe.InvoiceAddLinesLineParams{
+						Description: discount.Description,
+						Amount:      lo.ToPtr(-discount.Amount.GetFixed()),
+						Quantity:    lo.ToPtr(int64(1)),
+						Period:      period,
+					})
+				} else {
+					// Update line item
+					delete(existingStripeLines, line.ExternalIDs.Invoicing)
+
+					result.AddLineExternalID(line.ID, line.ExternalIDs.Invoicing)
+
+					updateStripeLines.Lines = append(updateStripeLines.Lines, &stripe.InvoiceUpdateLinesLineParams{
+						ID:          lo.ToPtr(line.ExternalIDs.Invoicing),
+						Description: discount.Description,
+						Amount:      lo.ToPtr(-discount.Amount.GetFixed()),
+						Quantity:    lo.ToPtr(int64(1)),
+						// period is not updatable
+					})
+				}
+			}
+		})
+
+		// Add line item
+		switch line.Type {
+		case billing.InvoiceLineTypeFee:
+			// Add line item if it doesn't exist
+			if line.ExternalIDs.Invoicing == "" {
+				lineIDs = append(lineIDs, line.ID)
+
+				addStripeLines.Lines = append(addStripeLines.Lines, &stripe.InvoiceAddLinesLineParams{
+					Description: line.Description,
+					Amount:      lo.ToPtr(line.Totals.Amount.GetFixed()),
+					Quantity:    lo.ToPtr(int64(1)),
+					Period:      period,
+				})
+			} else {
+				// Update line item
+				delete(existingStripeLines, line.ExternalIDs.Invoicing)
+
+				result.AddLineExternalID(line.ID, line.ExternalIDs.Invoicing)
+
+				updateStripeLines.Lines = append(updateStripeLines.Lines, &stripe.InvoiceUpdateLinesLineParams{
+					ID:          lo.ToPtr(line.ExternalIDs.Invoicing),
+					Description: line.Description,
+					Amount:      lo.ToPtr(line.Totals.Amount.GetFixed()),
+					Quantity:    lo.ToPtr(int64(1)),
+					// period is not updatable
+				})
+			}
+
+		case billing.InvoiceLineTypeUsageBased:
+			// Add line item if it doesn't exist
+			if line.ExternalIDs.Invoicing == "" {
+				lineIDs = append(lineIDs, line.ID)
+
+				addStripeLines.Lines = append(addStripeLines.Lines, &stripe.InvoiceAddLinesLineParams{
+					Description: line.Description,
+					Amount:      lo.ToPtr(line.Totals.Amount.GetFixed()),
+					Quantity:    lo.ToPtr(line.UsageBased.Quantity.GetFixed()),
+					Period:      period,
+				})
+			} else {
+				// Update line item
+				delete(existingStripeLines, line.ExternalIDs.Invoicing)
+
+				result.AddLineExternalID(line.ID, line.ExternalIDs.Invoicing)
+
+				updateStripeLines.Lines = append(updateStripeLines.Lines, &stripe.InvoiceUpdateLinesLineParams{
+					ID:          lo.ToPtr(line.ExternalIDs.Invoicing),
+					Description: line.Description,
+					Amount:      lo.ToPtr(line.Totals.Amount.GetFixed()),
+					Quantity:    lo.ToPtr(line.UsageBased.Quantity.GetFixed()),
+					// period is not updatable
+				})
```

Comment:
If you are syncronizing the UsageBasedLine and it's children the total will be doubled. Either synchronize the children only or the UBP line's totals but both both.